spec/spec.md

```diff
@@ -0,0 +1,142 @@
+
+1. **URI**: a URI that follows the [RFC 3986 specification](https://datatracker.ietf.org/doc/html/rfc3986), e.g. `s3://mu-bucket/prefix`.
+2. **Absolute POSIX storage path**: an absolute file path in a POSIX standard storage, e.g. `/my/dir`.
+3. **Relative POSIX storage path**: a relative file path in a POSIX standard storage, e.g. `my/dir2`, `./my/dir3`.
+   The absolute path of the directory should be based on the current directory of the running process.
+
+#### Table Existence
+
+A table exists in a Lance directory catalog if a table directory of the specific name exists.
+This is true even if the directory is empty or the contents in the directory does not follow the Lance table format spec.
+For such cases, an operation that lists all tables in the directory should show the specific table,
+and an operation that checks if a table exists should return true.
+However, an operation that loads the Lance table metadata should fail with error 
+indicating the content in the folder is not compliant with the Lance table format spec.
+
+### Lance REST Catalog
+
+In an enterprise environment, typically there is a requirement to store tables in a catalog service 
+such as Apache Hive MetaStore, Apache Gravitino, Unity Catalog, etc. 
+for more advanced governance features around access control, auditing, lineage tracking, etc.
+**Lance REST catalog** is an OpenAPI protocol that enables reading, writing and managing Lance tables
+by connecting those catalog services or building a custom catalog server in a standardized way.
+The detailed OpenAPI specification content can be found in [rest-catalog.yaml](./rest-catalog.yaml).
+
+#### Catalog Server and Adapter
+
+Any REST HTTP server that implements this OpenAPI protocol is called a **Lance Catalog Server**.
+If the main purpose of this server is to be a proxy on top of an existing catalog service,
+converting back and forth between Lance REST API models and native API models of the catalog service,
+then this Lance catalog server is called a **Lance Catalog Adapter**.
+
+#### Relationship with Lance Table on Storage
+
+This section defines the possible relationships between a Lance table definition in a REST catalog and
+the corresponding Lance table sitting in some storage system.
+A catalog can choose to support one or multiple types of relationships when integrating with the REST protocol.
+
+##### Catalog Managed Table
+
+A catalog managed Lance table is a table that is fully managed by a catalog.
+The catalog must contain information about the latest version of the Lance table.
+Any modifications to the table must happen through the catalog.
+If a user directly modifies the underlying table in the storage bypassing catalog,
+the catalog must not reflect the changes in the table to the catalog users.
+
+This mode ensures the catalog service is aware of all activities in the table,
+and can thus fully enforce any governance and management features for the table. 
+
+##### Storage Managed Table
```

Comment:
What about naming `native table`? That means directly using the native SDK-created tables.